src/aosm/azext_aosm/_configuration.py

```diff
@@ -59,9 +62,59 @@ def validate(self):
         if not self.version:
             raise ValidationError("version must be set.")
         if self.blob_sas_url and self.file_path:
-            raise ValidationError("Only one of file_path or blob_sas_url may be set.")
+            raise ValidationError(
+                "Only one of file_path or blob_sas_url may be set."
+            )
         if not (self.blob_sas_url or self.file_path):
-            raise ValidationError("One of file_path or sas_blob_url must be set.")
+            raise ValidationError(
+                "One of file_path or sas_blob_url must be set."
+            )
+
+
+@dataclass
+class VhdArtifactConfig(ArtifactConfig):
+    image_disk_size_GB: Optional[Union[str, int]] = None
+    image_os_state: Optional[str] = None
+    image_os_type: Optional[str] = None
+    image_hyper_v_generation: Optional[str] = None
+    image_api_version: Optional[str] = None
+
+    def __post_init__(self):
+        """
+        Convert parameters to the correct types.
+        """
+        if (
+            isinstance(self.image_disk_size_GB, str)
+            and self.image_disk_size_GB.isdigit()
+        ):
+            self.image_disk_size_GB = int(self.image_disk_size_GB)
+
+    @classmethod
+    def helptext(cls) -> "VhdArtifactConfig":
+        """
+        Build an object where each value is helptext for that field.
+        """
+        return VhdArtifactConfig(
+            image_disk_size_GB=(
+                "Optional. Specifies the size of empty data disks in gigabytes. "
+                "This value cannot be larger than 1023 GB."
+            ),
+            image_os_state=(
+                "Optional. The OS State. For managed images, use Generalized."
+            ),
+            image_os_type=(
+                "Optional. This property allows you to specify the type of the "
+                "OS that is included in the disk if creating a VM from a custom image."
```

Comment:
Ditto - aren't we always using custom images in AOSM? (I'm not sure). This is taken direct from https://learn.microsoft.com/en-us/azure/templates/microsoft.compute/images?pivots=deployment-language-bicep#imageosdisk

Reply:
https://learn.microsoft.com/en-us/azure/virtual-machines/image-version?tabs=portal%2Ccli2 might help explain what a custom image is.

Reply:
Allowed values are windows and linux, the default is linux if not specified.